public/app/features/alerting/unified/testSetup/datasources.ts

```diff
@@ -2,15 +2,19 @@ import { keyBy } from 'lodash';
 
 import { DataSourceInstanceSettings } from '@grafana/data';
 import { config, setDataSourceSrv } from '@grafana/runtime';
+import { DatasourceSrv } from 'app/features/plugins/datasource_srv';
 
-import { MockDataSourceSrv } from '../mocks';
+// import { MockDataSourceSrv } from '../mocks';
 
 /**
  * Sets up the data sources for the tests.
  * Sets up both config object from grafana/runtime and the data source server
  * @param configs data source instance settings. Use **mockDataSource** to create mock settings
  */
 export function setupDataSources(...configs: DataSourceInstanceSettings[]) {
-  config.datasources = keyBy(configs, (c) => c.name);
```

Comment:
Setting up the datasources like this means that, within datasource srv, when we try and fetch a full datasource with its components, the property is already set, so it skips setting up the full datasource config

i.e. it doesn't go into the `loadDatasource` method, which is what imports the plugin components

https://github.com/grafana/grafana/blob/e93cbc361a39de0fa84a946652f7162e8c7472dc/public/app/features/plugins/datasource_srv.ts#L129-L131

This change seems to work for all but one of the existing tests in the unified alerting folder. That one test ends up with a max call stack exceeded, which I haven't investigated properly yet